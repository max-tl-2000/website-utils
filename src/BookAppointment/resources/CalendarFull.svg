<?xml version="1.0" encoding="utf-8"?>
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="40" height="40" viewBox="0 0 40 40">
  <path d="M8.27586 0C3.71241 0 0 3.71241 0 8.27586C0 12.8393 3.71241 16.5517 8.27586 16.5517C12.8393 16.5517 16.5517 12.8393 16.5517 8.27586C16.5517 3.71241 12.8393 0 8.27586 0ZM8.27586 15.1724C4.4731 15.1724 1.37931 12.0786 1.37931 8.27586C1.37931 4.4731 4.4731 1.37931 8.27586 1.37931C12.0786 1.37931 15.1724 4.4731 15.1724 8.27586C15.1724 12.0786 12.0786 15.1724 8.27586 15.1724Z" transform="translate(23.4483 23.4482)" fill-opacity="0.27"/>
  <path d="M4.82759 4.13793H1.37931V0.689655C1.37931 0.308276 1.07103 0 0.689655 0C0.308276 0 0 0.308276 0 0.689655V4.82759C0 5.20897 0.308276 5.51724 0.689655 5.51724H4.82759C5.20897 5.51724 5.51724 5.20897 5.51724 4.82759C5.51724 4.44621 5.20897 4.13793 4.82759 4.13793Z" transform="translate(34.658 27.8423) rotate(45)" fill-opacity="0.27"/>
  <path d="M4.82759 4.13793H1.37931V0.689655C1.37931 0.308276 1.07103 0 0.689655 0C0.308276 0 0 0.308276 0 0.689655V4.82759C0 5.20897 0.308276 5.51724 0.689655 5.51724H4.82759C5.20897 5.51724 5.51724 5.20897 5.51724 4.82759C5.51724 4.44621 5.20897 4.13793 4.82759 4.13793Z" transform="translate(28.7516 35.645) rotate(-134.794)" fill-opacity="0.27"/>
  <path d="M13.7931 0H12.4138H7.58621H6.2069H0V6.2069V7.58621V12.4138V13.7931V20H6.2069H7.58621H13.7931V13.7931V12.4138V7.58621H18.6207H20H26.2069V0H20H18.6207H13.7931ZM20 1.37931H24.8276V6.2069H20V1.37931ZM7.58621 1.37931H12.4138V6.2069H7.58621V1.37931ZM1.37931 1.37931H6.2069V6.2069H1.37931V1.37931ZM1.37931 7.58621H6.2069V12.4138H1.37931V7.58621ZM6.2069 18.6207H1.37931V13.7931H6.2069V18.6207ZM12.4138 18.6207H7.58621V13.7931H12.4138V18.6207ZM12.4138 12.4138H7.58621V7.58621H12.4138V12.4138ZM18.6207 6.2069H13.7931V1.37931H18.6207V6.2069Z" transform="translate(4.82758 14.4829)" fill-opacity="0.27"/>
  <path d="M22.7586 37.931H1.37931V11.0345H34.4828V20C34.4828 20.3814 34.791 20.6897 35.1724 20.6897C35.5538 20.6897 35.8621 20.3814 35.8621 20V10.3448V3.44828C35.8621 3.0669 35.5538 2.75862 35.1724 2.75862H31.7241V0.689655C31.7241 0.308276 31.4159 0 31.0345 0H26.2069C25.8255 0 25.5172 0.308276 25.5172 0.689655V2.75862H10.3448V0.689655C10.3448 0.308276 10.0366 0 9.65517 0H4.82759C4.44621 0 4.13793 0.308276 4.13793 0.689655V2.75862H0.689655C0.308276 2.75862 0 3.0669 0 3.44828V10.3448V38.6207C0 39.0021 0.308276 39.3103 0.689655 39.3103H22.7586C23.14 39.3103 23.4483 39.0021 23.4483 38.6207C23.4483 38.2393 23.14 37.931 22.7586 37.931ZM26.8966 1.37931H30.3448V3.44828V5.51724H26.8966V3.44828V1.37931ZM5.51724 1.37931H8.96552V3.44828V5.51724H5.51724V3.44828V1.37931ZM1.37931 4.13793H4.13793V6.2069C4.13793 6.58828 4.44621 6.89655 4.82759 6.89655H9.65517C10.0366 6.89655 10.3448 6.58828 10.3448 6.2069V4.13793H25.5172V6.2069C25.5172 6.58828 25.8255 6.89655 26.2069 6.89655H31.0345C31.4159 6.89655 31.7241 6.58828 31.7241 6.2069V4.13793H34.4828V9.65517H1.37931V4.13793Z" fill-opacity="0.27"/>
</svg>
