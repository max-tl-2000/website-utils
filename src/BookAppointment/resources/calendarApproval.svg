<?xml version="1.0" encoding="utf-8"?>
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="40" height="40" viewBox="0 0 60 60">
  <path d="M12.4138 0C5.56863 0 0 5.56863 0 12.4138C0 19.259 5.56863 24.8276 12.4138 24.8276C19.259 24.8276 24.8276 19.259 24.8276 12.4138C24.8276 5.56863 19.259 0 12.4138 0ZM12.4138 22.7586C6.70966 22.7586 2.06897 18.1179 2.06897 12.4138C2.06897 6.70966 6.70966 2.06897 12.4138 2.06897C18.1179 2.06897 22.7586 6.70966 22.7586 12.4138C22.7586 18.1179 18.1179 22.7586 12.4138 22.7586Z" transform="translate(35.1724 35.1724)" />
  <path d="M20.6897 0H18.6207H11.3793H9.31035H0V9.31035V11.3793V18.6207V20.6897V30H9.31035H11.3793H20.6897V20.6897V18.6207V11.3793H27.9311H30H39.3104V0H30H27.9311H20.6897ZM30 2.06897H37.2414V9.31035H30V2.06897ZM11.3793 2.06897H18.6207V9.31035H11.3793V2.06897ZM2.06897 2.06897H9.31035V9.31035H2.06897V2.06897ZM2.06897 11.3793H9.31035V18.6207H2.06897V11.3793ZM9.31035 27.9311H2.06897V20.6897H9.31035V27.9311ZM18.6207 27.9311H11.3793V20.6897H18.6207V27.9311ZM18.6207 18.6207H11.3793V11.3793H18.6207V18.6207ZM27.9311 9.31035H20.6897V2.06897H27.9311V9.31035Z" transform="translate(7.24146 21.7241)" />
  <path d="M34.138 56.8966H2.06897V16.5517H51.7242V30C51.7242 30.5721 52.1866 31.0345 52.7587 31.0345C53.3307 31.0345 53.7932 30.5721 53.7932 30V15.5173V5.17242C53.7932 4.60035 53.3307 4.13793 52.7587 4.13793H47.5863V1.03448C47.5863 0.462414 47.1238 0 46.5518 0H39.3104C38.7383 0 38.2759 0.462414 38.2759 1.03448V4.13793H15.5173V1.03448C15.5173 0.462414 15.0548 0 14.4828 0H7.24139C6.66932 0 6.2069 0.462414 6.2069 1.03448V4.13793H1.03448C0.462414 4.13793 0 4.60035 0 5.17242V15.5173V57.9311C0 58.5032 0.462414 58.9656 1.03448 58.9656H34.138C34.71 58.9656 35.1724 58.5032 35.1724 57.9311C35.1724 57.359 34.71 56.8966 34.138 56.8966ZM40.3449 2.06897H45.5173V5.17242V8.27587H40.3449V5.17242V2.06897ZM8.27587 2.06897H13.4483V5.17242V8.27587H8.27587V5.17242V2.06897ZM2.06897 6.2069H6.2069V9.31035C6.2069 9.88242 6.66932 10.3448 7.24139 10.3448H14.4828C15.0548 10.3448 15.5173 9.88242 15.5173 9.31035V6.2069H38.2759V9.31035C38.2759 9.88242 38.7383 10.3448 39.3104 10.3448H46.5518C47.1238 10.3448 47.5863 9.88242 47.5863 9.31035V6.2069H51.7242V14.4828H2.06897V6.2069Z" />
  <path d="M4.21705 6.18561L1.4961 3.56557L0 5.0062L3.59064 8.4637L3.59476 8.4597L4.25027 9.09091L12.7273 1.46863L11.2022 0L4.21705 6.18561Z" transform="translate(41.8184 43.6362)" />
</svg>
