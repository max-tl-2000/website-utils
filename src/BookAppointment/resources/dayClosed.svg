<?xml version="1.0" encoding="utf-8"?>
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="40" height="40" viewBox="0 0 40 40">
  <path d="M20 0C8.97167 0 0 8.97223 0 20.0006C0.000928264 31.0278 8.97297 39.9994 20 39.9994C31.028 39.9994 40 31.028 40 20.0004C39.9998 8.97223 31.028 0 20 0ZM20 2.78479C24.0883 2.78479 27.8474 4.21822 30.8035 6.60775L6.60794 30.8031C4.21877 27.8472 2.78516 24.0883 2.78479 20.0004C2.78479 10.5078 10.5076 2.78479 20 2.78479ZM20 37.2146C15.5967 37.2146 11.5749 35.5519 8.52722 32.8225L32.8227 8.52703C35.5523 11.5747 37.215 15.5969 37.215 20.0006C37.215 29.4924 29.4922 37.2146 20 37.2146Z" fill-opacity="0.27"/>
</svg>

