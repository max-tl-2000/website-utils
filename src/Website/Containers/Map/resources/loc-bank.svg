<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 23.0.3, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 24 24" style="enable-background:new 0 0 24 24;" xml:space="preserve">
<path d="M11.5,1L2,6v2h19V6L11.5,1z M16,10v7h3v-7H16z M2,22h19v-3H2V22z M10,10v7h3v-7H10z M4,10v7h3v-7H4z"/>
</svg>
