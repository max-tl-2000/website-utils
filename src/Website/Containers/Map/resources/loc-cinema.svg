<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 23.0.3, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 24 24" style="enable-background:new 0 0 24 24;" xml:space="preserve">
<path d="M4,15h2c1.1,0,2,0.9,2,2v2h1v-2c0-1.1,0.9-2,2-2h2c1.1,0,2,0.9,2,2v2h1v-2c0-1.1,0.9-2,2-2h2c1.1,0,2,0.9,2,2v2h1v3H1v-3h1
	v-2C2,15.9,2.9,15,4,15z M11,7l4,3l-4,3V7z M4,2h16c1.1,0,2,0.9,2,2v9.54C21.41,13.19,20.73,13,20,13V4H4v9
	c-0.73,0-1.41,0.19-2,0.54V4C2,2.9,2.9,2,4,2z"/>
</svg>
