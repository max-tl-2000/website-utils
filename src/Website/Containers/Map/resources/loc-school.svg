<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 23.0.3, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 24 24" style="enable-background:new 0 0 24 24;" xml:space="preserve">
<path d="M12,3L1,9l11,6l9-4.91V17h2V9L12,3z M5,13.18v4L12,21l7-3.82v-4L12,17L5,13.18z"/>
</svg>
