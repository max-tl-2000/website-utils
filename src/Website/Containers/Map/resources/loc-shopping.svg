<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 23.0.3, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 24 24" style="enable-background:new 0 0 24 24;" xml:space="preserve">
<path d="M12,13c-2.76,0-5-2.24-5-5h2c0,1.66,1.34,3,3,3s3-1.34,3-3h2C17,10.76,14.76,13,12,13z M12,3c1.66,0,3,1.34,3,3H9
	C9,4.34,10.34,3,12,3z M19,6h-2c0-2.76-2.24-5-5-5S7,3.24,7,6H5c-1.11,0-1.99,0.9-1.99,2L3,20c0,1.1,0.89,2,2,2h14c1.1,0,2-0.9,2-2
	V8C21,6.89,20.1,6,19,6z"/>
</svg>
