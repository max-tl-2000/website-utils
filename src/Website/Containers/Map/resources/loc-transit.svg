<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 23.0.3, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 24 24" style="enable-background:new 0 0 24 24;" xml:space="preserve">
<path d="M12,4H5C3.3,4,2,5.3,2,7v8c0,1.7,1.3,3,3,3l-1,1v1h1l2-2l2,0v-5H4V6l9,0v2h2V7C15,5.3,13.7,4,12,4z M5,14c0.6,0,1,0.4,1,1
	s-0.4,1-1,1s-1-0.4-1-1S4.4,14,5,14z M22,13.8l-1.4-4.1C20.4,9.3,20,9,19.6,9h-7.2c-0.5,0-0.8,0.3-1,0.7L10,13.8l0,5.5
	c0,0.4,0.3,0.7,0.7,0.7h0.6c0.4,0,0.7-0.4,0.7-0.8V18h8v1.2c0,0.4,0.3,0.8,0.7,0.8h0.6c0.4,0,0.7-0.3,0.7-0.7L22,13.8z M12.4,10h7.2
	l1,3h-9.2L12.4,10z M12,16c-0.6,0-1-0.4-1-1s0.4-1,1-1s1,0.4,1,1S12.6,16,12,16z M20,16c-0.5,0-1-0.4-1-1s0.5-1,1-1s1,0.4,1,1
	S20.5,16,20,16z"/>
</svg>
