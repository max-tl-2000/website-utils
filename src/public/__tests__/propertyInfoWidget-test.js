/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

/* eslint-disable global-require */
import { createSandboxDiv, removeSandBoxDiv } from '../../../resources/sandbox';
import { mockWrap } from '../../../resources/mocker';

const { mockModules } = mockWrap(jest);

describe('propertyInfoWidget', () => {
  let createPropertyInfoWidget;
  beforeEach(() => {
    mockModules({
      '../../Website/Stores/WebSiteStore': {
        getWebSiteStore: () => ({}),
      },
      '../../Models/GuestCardModel': {
        GuestCardModel: () => ({}),
      },
    });
    createPropertyInfoWidget = require('../propertyInfoWidget').createPropertyInfoWidget;
  });

  describe('when no selector is provided', () => {
    it('should throw', () => {
      expect(() => createPropertyInfoWidget()).toThrowErrorMatchingSnapshot();
    });
  });

  describe('when the selector is not found in the dom', () => {
    it('should throw and mention the not found selector', () => {
      expect(() => createPropertyInfoWidget('#container')).toThrowErrorMatchingSnapshot();
    });
  });

  describe('when the selector is found but no template is found', () => {
    it('should throw complaining about missing template', () => {
      const { remove } = createSandboxDiv('container');

      expect(() => createPropertyInfoWidget('#container')).toThrowErrorMatchingSnapshot();

      remove();
    });
  });

  describe('when the selector is found and template is found', () => {
    let div;
    beforeEach(() => {
      div = createSandboxDiv('container').div;
    });

    afterEach(() => {
      removeSandBoxDiv('container');
    });

    it('should render the template inside the container and the template should be removed', () => {
      div.innerHTML = `
        <div id="test">
          <script type="text/x-twig">
            <p>Hello world</p>
          </script>
        </div>
      `;
      createPropertyInfoWidget('#test');
      const ele = div.querySelector('#test');
      expect(ele.querySelector('script[type="text/x-twig"]')).toBeNull();
      expect(ele.querySelector('p').innerHTML).toEqual('Hello world');
    });
  });
});
