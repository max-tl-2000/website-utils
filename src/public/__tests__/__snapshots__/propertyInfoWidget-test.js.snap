// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propertyInfoWidget when no selector is provided should throw 1`] = `"selector is required"`;

exports[`propertyInfoWidget when the selector is found but no template is found should throw complaining about missing template 1`] = `"missing template inside \\"#container\\""`;

exports[`propertyInfoWidget when the selector is not found in the dom should throw and mention the not found selector 1`] = `"selector: \\"#container\\" is not found"`;
